### DISCUSSION TITLE
How to assign a custom analyzer to a custom field?

### DISCUSSION THREAD

**Discussion by cadilhac** [ravendb/ravendb#16485] [Q&A] [answered]
I found some french analyzer code, generated its assembly and added it to a linux hosted ravendb server.

Here is my model and index code:

```
public class PostContent
{
    public string Title { get; set; }
    public string Excerpt { get; set; }
    public string Body { get; set; }
}

public class Post : RavenModel
{
    public Dictionary<string, PostContent> Content { get; set; } = new();
}

Map = posts => from post in posts
                select new
                {
                    _ = post.Content.Select(item => CreateField("Title_" + item.Key, item.Value.Title, false, true)),
                    __ = post.Content.Select(item => CreateField("Excerpt_" + item.Key, item.Value.Excerpt, false, true)),
                    ___ = post.Content.Select(item => CreateField("Body_" + item.Key, item.Value.Body, false, true))
                };

Analyzers.Add(x => "Title_fr", typeof(FrenchAnalyzer).AssemblyQualifiedName);
Analyzers.Add(x => "Excerpt_fr", typeof(FrenchAnalyzer).AssemblyQualifiedName);
Analyzers.Add(x => "Body_fr", typeof(FrenchAnalyzer).AssemblyQualifiedName);
```

As you can see, I am generating some custom fields for each language used in a blog post. In studio, I can successfully see that I have terms for Title_fr, Title_en, Excerpt_fr, and so on.
But I also see that no custom analyzers are used for these fields. It's as if Analyzers.Add had no effect. And I'm not sure about the syntax where it would accept a string expression as its first argument. Obviously, I can't use x => x.Title_fr since this field is dynamic.

How can I achieve this please? Thanks.

**Comment by DejanMilicic** [comment, marked as answer]
I have a working example, can you check it please?

https://github.com/DejanMilicic/RavenCustomAnalyzer/blob/main/Multilang/Search.cs

**Reply by cadilhac** [reply]
> I have a working example, can you check it please?
> 
> https://github.com/DejanMilicic/RavenCustomAnalyzer/blob/main/Multilang/Search.cs

Thank you. Using AnalyzersStrings instead of Analyzers works. Not sure why this is the one to use. I don't see it mentioned in the documentation.

**Reply by cadilhac** [reply]
However, I'm also using the same french analyzer as you and I noticed that it doesn't remove the l' and d' from the indexed terms. So if a string is "L'avenir de nos enfants", I can't search by "avenir" because it stored "l'avenir" as a term. Did you notice that? I have no idea how to fix that...

**Reply by ayende** [reply]
You may need to have a custom analyzer for that. Or customize the French one, I guess.
We use whatever Lucene has by default in this case.
